sqlglot/generator.py

```diff
@@ -1419,14 +1418,19 @@ def altertable_sql(self, expression: exp.AlterTable) -> str:
             actions = self.expressions(expression, "actions")
         elif isinstance(actions[0], exp.Delete):
             actions = self.expressions(expression, "actions", flat=True)
-        elif isinstance(actions[0], self.WITH_SINGLE_ALTER_TABLE_ACTION):
+        elif actions[0].__class__ in self.WITH_SINGLE_ALTER_TABLE_ACTION:
```

Comment:
why was this needed

Reply:
Not necessary to use, it's only helpful for reporting "unsupported" errors I think. We can probably put all cases under the `else` branch instead. Should I do that?